src/frontend/js/utils/routers/dashboard/index.tsx

```diff
@@ -0,0 +1,105 @@
+}
+
+// Translations of dashboard route paths
+const dashboardRoutePaths = defineMessages<DashboardPaths>({
+  [DashboardPaths.COURSES]: {
+    id: 'components.Dashboard.DashboardRoutes.courses.path',
+    description: 'The path to display the courses view.',
+    defaultMessage: '/courses',
+  },
+  [DashboardPaths.COURSE]: {
+    id: 'components.Dashboard.DashboardRoutes.course.path',
+    description: 'The path to display a course detail view.',
+    defaultMessage: '/courses/{code}',
+  },
+  [DashboardPaths.PREFERENCES]: {
+    id: 'components.Dashboard.DashboardRoutes.preferences.path',
+    description: 'The path to display the preferences view.',
+    defaultMessage: '/preferences',
+  },
+  [DashboardPaths.PREFERENCES_ADDRESS_EDITION]: {
+    id: 'components.Dashboard.DashboardRoutes.preferences.addresses.edition.path',
+    description: 'The path to display the addresses edition view.',
+    defaultMessage: '/preferences/addresses/{addressId}',
+  },
+  [DashboardPaths.PREFERENCES_CREDIT_CARD_EDITION]: {
+    id: 'components.Dashboard.DashboardRoutes.preferences.creditCards.edition.path',
+    description: 'The path to display the credit cards edition view.',
+    defaultMessage: '/preferences/credit-cards/{creditCardId}',
+  },
+});
+
+// Translations of dashboard route labels
+const dashboardRouteLabels = defineMessages<DashboardPaths>({
+  [DashboardPaths.COURSES]: {
+    id: 'components.Dashboard.DashboardRoutes.courses.label',
+    description: 'Label of the courses view used in navigation components.',
+    defaultMessage: 'My Courses',
+  },
+  [DashboardPaths.COURSE]: {
+    id: 'components.Dashboard.DashboardRoutes.course.session.label',
+    description: 'Label of the course session view used in navigation components.',
+    defaultMessage: 'Course',
+  },
+  [DashboardPaths.PREFERENCES]: {
+    id: 'components.Dashboard.DashboardRoutes.preferences.label',
+    description: 'Label of the preferences view used in navigation components.',
+    defaultMessage: 'My preferences',
+  },
+  [DashboardPaths.PREFERENCES_ADDRESS_EDITION]: {
+    id: 'components.Dashboard.DashboardRoutes.preferences.addresses.edition.label',
+    description: 'Label of the addresses edition view.',
+    defaultMessage: 'Billing addresses',
+  },
+  [DashboardPaths.PREFERENCES_CREDIT_CARD_EDITION]: {
+    id: 'components.Dashboard.DashboardRoutes.preferences.creditCards.label',
+    description: 'Label of the addresses edition view.',
+    defaultMessage: 'Credit cards',
+  },
+});
+
+/**
+ * Use `intl.formatMessage` to retrieve a path or label of a given route for the active locale.
+ *
+ * Currying function which takes first the type of attribute to translate (path or label)
+ * then takes the intl object retrieved from the IntlProvider.
+ * Finally, it takes a path and options to pass to `intl.formatMessage` method
+ */
+const getDashboardRouteAttribute =
+  (attribute: 'label' | 'path') =>
```

Comment:
```suggestion
  (attribute: 'label' | 'path') =>
```
maybe we could use const here

Reply:
Indeed I could use an enum.